pkg/plugins/golang/deploy-image/v1alpha1/scaffolds/internal/templates/controllers/controller.go

```diff
@@ -88,6 +89,13 @@ import (
 
 const {{ lower .Resource.Kind }}Finalizer = "{{ .Resource.Group }}.{{ .Resource.Domain }}/finalizer"
 
+// Definitions to manage status conditions
+const (
+	typeAvailable{{ .Resource.Kind }} = "Available"
+	typeProgressing{{ .Resource.Kind }} = "Progressing"
+	typeDegraded{{ .Resource.Kind }} = "Degraded"
+)
```

Comment:
@NikhilSharmaWe : 

```suggestion
const (
	// typeAvailable{{ .Resource.Kind }} represents the status of the Deployment reconciliation
	typeAvailable{{ .Resource.Kind }} = "Available"
	// typeDegraded{{ .Resource.Kind }} represents the status used when the custom resource is deleted
	// and the finalizer operations must to occurs. 
	typeDegraded{{ .Resource.Kind }} = "Degraded"
)
```

We can have only 2 contionalTypes. 

See: 
> Conditions provide a standard mechanism for higher-level status reporting from a controller. They are an extension mechanism which allows tools and other controllers to collect summary information about resources without needing to understand resource-specific status details. Conditions should complement more detailed information about the observed status of an object written by a controller, rather than replace it. For example, the "Available" condition of a Deployment can be determined by examining readyReplicas, replicas, and other properties of the Deployment. However, the "Available" condition allows other components to avoid duplicating the availability logic in the Deployment controller.

Available -. Status True would mean The Deployment is available and with the expected replicas
Available -  Status False is when it has an error and it is not available yet or is not with the specific size replicase
Available - Without status is when we begin the reconciliation and we do have not the Available donation so far
Degraded - we should add with no status when we start the finalizer and we should update with status true when we finish the finalise 

More info: https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#typical-status-properties